runtimes/protocol/chat.ts

```diff
@@ -59,19 +62,23 @@ export interface QuickActionsOptions {
     quickActionsCommandGroups: QuickActionCommandGroup[]
 }
 
-export interface ChatRequest extends ChatParams {
-    partialResultToken?: ProgressToken
-}
-
-export const chatRequestType = new ProtocolRequestType<ChatRequest, ChatResult, ChatResult, void, void>(
-    CHAT_REQUEST_METHOD
-)
+export const chatRequestType = new AutoParameterStructuresProtocolRequestType<
```

Comment:
Is this necessary? Can we map to `EncyptedChatParams` inside the `EncryptedChat`  class without defining a union type on the request?

Reply:
I think you are right, let me check